File: namegen.js

var fs = require('fs');

var args = process.argv.slice(2);
console.log('Function arguments: ', args);

var maxSyllables = args[0]
var minSyllables = args[1]
var numNames = args[2]
var sourceFile = args[3]

var names = new Array()

fs.readFile(sourceFile, function(err, data) {
    var sourceSyllables = data.toString().split("\r\n")
    console.log(`Number of syllable choices is ${sourceSyllables.length}`)
    for (j=0; j<numNames; j++) {
        var name = ""
        var numSyllables = getRandomIntInRange(minSyllables, maxSyllables)
        for (i=0; i<numSyllables;i++) {
            var sy = sourceSyllables[getRandomInt(sourceSyllables.length)].toString()
            name = name.concat("",sy.toString())
        }
        names.push(name)
    }
    console.log("Generated names:")
    names.forEach(n => {
        console.log(n)
    });
});


// var contents = fs.readFileSync(sourceFile, 'utf8');

// var sourceSyllables = contents.toString().split("\r\n")
// console.log(`Number of syllable choices is ${sourceSyllables.length}`)
// for (j=0; j<numNames; j++) {
//     var name = ""
//     var numSyllables = getRandomIntInRange(minSyllables, maxSyllables)
//     for (i=0; i<numSyllables;i++) {
//         var sy = sourceSyllables[getRandomInt(sourceSyllables.length)].toString()
//         name = name.concat("",sy.toString())
//     }
//     names.push(name)
// }
// console.log("Generated names:")
// names.forEach(n => {
//     console.log(n)
// });

function getRandomInt(max) {
    var rand = Math.floor(Math.random() * Math.floor(max));
    return rand
}

function getRandomIntInRange(min, max) {
    min = Math.ceil(min);
    max = Math.floor(max);
    var rand = Math.floor(Math.random() * (max - min) + min);
    return rand
}

File: namegen.py
import sys
import random

print 'Argument List:', str(sys.argv)

if (len(sys.argv) != 5):
    print("Usage:")
    print("python namegen.py <maximum syllables> <minimum syllables> <number of names> <syllables source file>")
    exit()

maxSyllables = int(sys.argv[1])
minSyllables = int(sys.argv[2])
numNames = int(sys.argv[3])
sourceFile = sys.argv[4]

f = open('dwarf.txt', 'r')
contents = f.read()

sourceSylables = contents.split("\n")

names = []

for i in range(0,numNames):
    name = ""
    numSyllables = random.randint(minSyllables,maxSyllables)
    for j in range(0,numSyllables):
        sy = sourceSylables[random.randint(0,len(sourceSylables))]
        name += sy
        # print 'my random sylable:',sourceSylables[random.randint(0,len(sourceSylables))]
    names.append(name)

print("Generated names:")
for i in names:
    print(i)